Add Usage and Behavior page

mcarrano opened the pull request:
Adds general Usage and behavior guidance to the Guidelines section of the website.

patternfly-build commented:
PF4 preview: https://patternfly-org-pr-2371-v4.surge.sh/v4

mcarrano commented:
@redallen I have some questions about this when you get a chance. If you look at the website preview, you'll see that some of the table headings get cut off. That's not the case when I preview the same markdown text in either GitHub or my desktop editor. That implies that it's something to do with the table presentation on the website. @mmenestr has run into similar issues and has made things work by adding extra spaces, but that does not seem like a robust solution. Any idea why this is happening and what we'd need to do to fix it? Ideally long headings would wrap when they are too wide for the column.

redallen commented:
There is no standard MD way to [add classes that we need to control text.](https://patternfly.org/v4/components/table/html#controlling-text) That means the only way to not have those columns cut off is to write the HTML yourself like:
```html
<table class="ws-table pf-c-table pf-m-grid-lg" role="grid">
   <thead>
      <tr role="row">
         <th role="columnheader" scope="col">
            <div></div>
         </th>
         <th align="center" role="columnheader" scope="col">
            <div><a href="/v4/components/select">Select</a></div>
         </th>
         <th align="center" role="columnheader" scope="col">
            <div><a href="/v4/components/options-menu">Options menu</a></div>
         </th>
         <th align="center" role="columnheader" scope="col">
            <div><a href="/v4/components/dropdown">Dropdown</a></div>
         </th>
      </tr>
   </thead>
   <tbody role="rowgroup">
      <tr role="row">
         <td role="cell">
            <div>You want to select a value or multiple values from a list.</div>
         </td>
         <td align="center" role="cell" data-label="Select">
            <div>✔</div>
         </td>
         <td align="center" role="cell" data-label="Options menu">
            <div>⛔</div>
         </td>
         <td align="center" role="cell" data-label="Dropdown">
            <div>⛔</div>
         </td>
      </tr>
      <tr role="row">
         <td role="cell">
            <div>You want to select filter terms from a list.</div>
         </td>
         <td align="center" role="cell" data-label="Select">
            <div>✔</div>
         </td>
         <td align="center" role="cell" data-label="Options menu">
            <div>⛔</div>
         </td>
         <td align="center" role="cell" data-label="Dropdown">
            <div>⛔</div>
         </td>
      </tr>
      <tr role="row">
         <td role="cell">
            <div>You want to make the selected value visible when the menu is closed.</div>
         </td>
         <td align="center" role="cell" data-label="Select">
            <div>✔</div>
         </td>
         <td align="center" role="cell" data-label="Options menu">
            <div>⛔</div>
         </td>
         <td align="center" role="cell" data-label="Dropdown">
            <div>⛔</div>
         </td>
      </tr>
      <tr role="row">
         <td role="cell">
            <div>You want to select options from one or more lists (sorting options, for example).</div>
         </td>
         <td align="center" role="cell" data-label="Select">
            <div>⛔</div>
         </td>
         <td align="center" role="cell" data-label="Options menu">
            <div>✔</div>
         </td>
         <td align="center" role="cell" data-label="Dropdown">
            <div>⛔</div>
         </td>
      </tr>
      <tr role="row">
         <td role="cell">
            <div>You want to expose a list of commands or actions in a limited space.</div>
         </td>
         <td align="center" role="cell" data-label="Select">
            <div>⛔</div>
         </td>
         <td align="center" role="cell" data-label="Options menu">
            <div>⛔</div>
         </td>
         <td align="center" role="cell" data-label="Dropdown">
            <div>✔</div>
         </td>
      </tr>
   </tbody>
</table>
```
and then add the text controlling classes according to the docs.

mcarrano commented:
@mmenestr this is now ready for final review, please take a look when you can.

@redallen @andybraren thanks for working out the problem with table header wrapping. That has resolved the problems I was previously having with this.

mmenestr commented on packages/v4/src/content/design-guidelines/usage-and-behavior/usage-and-behavior.md:
```diff
@@ -0,0 +1,119 @@
+
+Tables are built from a standard tabular structure of rows and columns. Data lists support any valid HTML layout inside of a row and therefore allow more formatting flexibility.
+
+|                         | [Table](/components/table) | [Data list](/components/data-list) |
```
Feels like the left column should have a header.. like "Use case" or "Scenario"

mcarrano replied:
@afialkoff had a similar comment when she reviewed the Google doc version. I did not feel a title here would add much, but can you take a look at this again now that it's viewable in the website preview @afialkoff ?  If you both think it will add value then I'm good to add a heading above the first column.

afialkoff replied:
Hm... After looking at the preview, my gut says adding that heading would be valuable — if only to balance out the top row. The devil's in the details, and while I don't think anyone would get confused as to what the first column communicates, I could see potential for some readers to get distracted by that empty cell.

Of those two options @mmenestr supplied, I think "Use case" would work the best above the first column. It's short and reflects rhetoric we use regularly throughout our docs. 👍

mmenestr commented on packages/v4/src/content/design-guidelines/usage-and-behavior/usage-and-behavior.md:
```diff
@@ -0,0 +1,119 @@
+---
+
+## Displaying structured data
+Structured data includes any information that is stored in a database or similar regular data structure. The most common ways of presenting structured data are to display it in either a list or a table where rows correspond to records in the database. PatternFly supports two controls for displaying this type of data: the [Data list](/components/data-list) and the [Table](/components/table).
```
You capitalize data list and table here but in the rest of the page when you refer to components, they are sentence cased! Think these should be sentence cased as well for consistency

mmenestr reviewed (commented):
Overall looks great, thanks for doing all this Matt! Just left 2 small comments, and otherwise the only thing I wonder is whether we should add a small intro at the top of the page to introudce the purpose of this page, like:

"These are usage and behavior guidelines to help you navigate and choose which PatternFly component to use depending on your use cases and scenarios." something along those lines!

mcarrano commented:
Good catch on the intro @mmenestr . I actually had that in the Google doc version but must have gotten lost in translation. I'll add it back.

mcarrano commented:
@mmenestr @afialkoff Thanks for you feedback. I addressed issues by adding a short intro and the heading "Use case" to the column on all tables. Please give this another review.

mmenestr reviewed (approved):
Looks great Matt, merging!
